### ISSUE TITLE
-r/-rL allow for manual specification of list of resolvers[feature] 

### ISSUE THREAD

**Issue by PR3R00T** [projectdiscovery/naabu#124]
Please add -r/-rL like dnsx to allow a user to specify a list of resolvers to use during the DNS lookup of parsed domains, this can allow for the scanner to scale without the issue of being blocked by a DNS server.

**Comment by bauthard** [comment]
Hi @PR3R00T, 

**Naabu** is not the right place where you should be focused on resolving part, the expected case is, you will have a list of resolved hosts/subs using `dnsx` or any of your preferred tool and you feed the list to **naabu** for port scanning, you should not be passing a passive list of subs directly to naabu, apart from that what use cases do you see on your side to have this feature in naabu itself. 

In general, we focus on one tool doing one task, so what you suggested can be accomplished with `dnsx -l subs.txt -r resolvers.txt | naabu`

let me know what you think about it.

**Comment by PR3R00T** [comment]
Yeah agreed however when you parse through the list of resolved domains from DNSX, unless I'm missing something, Naabu will need to resolve the hosts again before doing its port scan?

**Comment by bauthard** [comment]
> Naabu will need to resolve the hosts again before doing its port scan?

@PR3R00T right, it will resolve again which also strikes the idea to have a solution which will allow us to utilise the data from `dnsx` throughout the process using other PD tools, bascailly the the same data will be used from start to end in following manner `dnsx | naabu | httpx | nuclei` without repeating the host resolutions multiple times, and it will take some time as changes will be required in each project, for the time being, it's fair feature suggestion to support custom resolver input, and we will implement in next release.

**Comment by PR3R00T** [comment]
@bauthard Long term plan sounds great! Thanks :)

**Comment by alxjsn** [comment]
@ehsandeep the behavior here is unusual since there are hard-coded resolvers in the dnsx library being used instead of using system resolvers. I currently have mine patched to use system resolvers, but could this be the default?

**Comment by Mzack9999** [comment]
@PR3R00T @alxjsn Thanks for reporting the issue. The feature is already implemented in [fastdialer](https://github.com/projectdiscovery/fastdialer) where we parse the system resolvers and use them before the hardcoded ones. We will try to port the code to naabu as well, thanks for your patience!
